packages/ui/client/components/views/ViewEditor.vue

```diff
@@ -134,54 +139,108 @@ function createErrorElement(e: ErrorWithDiff) {
   const el: EventListener = async () => {
     await openInEditor(stack.file, stack.line, stack.column)
   }
+  span.addEventListener('click', el)
   div.appendChild(span)
   listeners.push([span, el, () => destroyTooltip(span)])
   handles.push(codemirrorRef.value!.addLineClass(stack.line - 1, 'wrap', 'bg-red-500/10'))
   widgets.push(codemirrorRef.value!.addLineWidget(stack.line - 1, div))
 }
 
-watch(
-  [cm, failed],
-  ([cmValue]) => {
+const { pause, resume } = watch(
+  [codemirrorRef, failed, finished, saving] as const,
```

Comment:
add hint here: the watcher receiving `saving` with true even when not changed, that's why we use pause and resume it in `onSave`.

The problem without pausing this watcher when saving the file => cursor position lost (the caret will go to the first line in the test)

Reply:
The watcher loading the file content being called multiple times when the server re-run the file, splitting he logic for loading and saving will prevent reloading the content while saving from the view editor.

The cursor will be preserved only when saving in the file in the ui view editor, saving the file in the IDE will lost the position (will use the line in the url)

Reply:
Note: we should lock the editor or check if there is something changed when saving, but we cannot set the editor in readonly (maybe refreshing  the editor works), rn you can save at any time using the keyboard.

I Will check it again later since latest commit will prevent reloading the race condition in loading watcher.

Reply:
looks like using `setOption('readOnly', true/false)` + `refresh()` we can prevent saving the file while saving it, updated logic and added protection to prevent firing `save` event when editor in readOnly mode (also for static html reporter)